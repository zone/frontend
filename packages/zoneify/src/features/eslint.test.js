const { addDependency, addPackageJsonChange } = require('../install')
const { name, questions, run } = require('./eslint')

jest.mock('../install')

describe('features/eslint', () => {
  it('has a name', () => {
    expect(name).toMatchSnapshot()
  })

  it('has questions', () => {
    expect(questions).toMatchSnapshot()
  })

  describe('run', () => {
    const testPackageJson = { name: 'test package' }

    it('has a run function', () => {
      expect(typeof run).toBe('function')
    })

    describe('no framework', () => {
      run({ framework: 'zone' })

      it('adds a dependency', () => {
        expect(addDependency).toBeCalledWith({
          name: '@zonedigital/eslint-config-zone',
          includePeerDependencies: true,
        })
      })

      it('adds a config change', () => {
        expect(addPackageJsonChange).toMatchSnapshot()
        expect(typeof addPackageJsonChange.mock.results[0].value).toBe(
          'function'
        )
        expect(
          addPackageJsonChange.mock.results[0].value(testPackageJson)
        ).toMatchSnapshot()
      })
    })

    describe('react', () => {
      run({ framework: 'react' })

      it('adds a dependency', () => {
        expect(addDependency).toBeCalledWith({
          name: '@zonedigital/eslint-config-zone',
          includePeerDependencies: true,
        })
      })

      it('adds a config change', () => {
        expect(addPackageJsonChange).toMatchSnapshot()
        expect(typeof addPackageJsonChange.mock.results[0].value).toBe(
          'function'
        )
        expect(
          addPackageJsonChange.mock.results[0].value(testPackageJson)
        ).toMatchSnapshot()
      })
    })

    describe('vue', () => {
      run({ framework: 'vue' })

      it('adds a dependency', () => {
        expect(addDependency).toBeCalledWith({
          name: '@zonedigital/eslint-config-zone',
          includePeerDependencies: true,
        })
      })

      it('adds a config change', () => {
        addPackageJsonChange.mockReturnValue(() => 'hello')
        expect(addPackageJsonChange).toMatchSnapshot()
        expect(typeof addPackageJsonChange.mock.results[0].value).toBe(
          'function'
        )
        expect(
          addPackageJsonChange.mock.results[0].value(testPackageJson)
        ).toMatchSnapshot()
      })
    })
  })
})
