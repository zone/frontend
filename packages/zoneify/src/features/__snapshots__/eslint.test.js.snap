// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/eslint has a name 1`] = `"ESLint"`;

exports[`features/eslint has questions 1`] = `
Array [
  Object {
    "choices": Array [
      Object {
        "name": "None",
        "value": "zone",
      },
      Object {
        "name": "React",
        "value": "react",
      },
      Object {
        "name": "Vue",
        "value": "vue",
      },
    ],
    "message": "What type of JS framework are you using?",
    "name": "framework",
    "type": "list",
  },
]
`;

exports[`features/eslint run no framework adds a config change 1`] = `
[MockFunction addPackageJsonChange] {
  "calls": Array [
    Array [
      [Function],
    ],
    Array [
      [Function],
    ],
    Array [
      [Function],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;

exports[`features/eslint run no framework adds a config change 2`] = `
Object {
  "eslintConfig": Object {
    "extends": "@zonedigital/eslint-config-zone",
  },
  "name": "test package",
}
`;

exports[`features/eslint run react adds a config change 1`] = `
[MockFunction addPackageJsonChange] {
  "calls": Array [
    Array [
      [Function],
    ],
    Array [
      [Function],
    ],
    Array [
      [Function],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;

exports[`features/eslint run react adds a config change 2`] = `
Object {
  "eslintConfig": Object {
    "extends": "@zonedigital/eslint-config-zone",
  },
  "name": "test package",
}
`;

exports[`features/eslint run vue adds a config change 1`] = `
[MockFunction addPackageJsonChange] {
  "calls": Array [
    Array [
      [Function],
    ],
    Array [
      [Function],
    ],
    Array [
      [Function],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;

exports[`features/eslint run vue adds a config change 2`] = `
Object {
  "eslintConfig": Object {
    "extends": "@zonedigital/eslint-config-zone",
  },
  "name": "test package",
}
`;
